src/main/scala/com/ubirch/filter/model/cache/CacheImpl.scala

```diff
@@ -6,68 +6,57 @@ import com.ubirch.filter.ConfPaths.RedisConfPaths._
 import com.ubirch.filter.services.Lifecycle
 import monix.execution.Scheduler
 import scredis.Redis
+import scredis.exceptions.RedisIOException
 import scredis.protocol.AuthConfig
 
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.Future
-import scala.concurrent.duration.DurationInt
 
 @Singleton
 class CacheImpl @Inject()(config: Config, lifecycle: Lifecycle)
                          (implicit scheduler: Scheduler) extends Cache with StrictLogging {
 
   private val host = config.getString(REDIS_MAIN_HOST)
   private val port = config.getInt(REDIS_PORT)
+  private val index = config.getInt(REDIS_INDEX)
+  private val cacheTTL = config.getInt(REDIS_CACHE_TTL)
+
   private val password: String = config.getString(REDIS_PASSWORD)
   private val authOpt =
     if (password == "") None
     else Some(AuthConfig(None, password))
-  private val index = config.getInt(REDIS_INDEX)
-
-  private val cacheTTL = config.getInt(REDIS_CACHE_TTL)
-  private var redis: Redis = _
 
-  private val initialDelay = 1.seconds
-  private val repeatingDelay = 2.seconds
-
-  private val c = scheduler.scheduleAtFixedRate(initialDelay, repeatingDelay) {
-    try {
-      redis = Redis(host, port, authOpt, index)
-      stopConnecting()
-      logger.info("connection to redis cache has been established.")
-    } catch {
-      case ex: Exception =>
-        logger.error("redis error: not able to create connection: ", ex.getMessage, ex)
-    }
-  }
-
-  private def stopConnecting(): Unit = {
-    c.cancel()
-  }
+  /**
+    * This redis instance doesn't connect yet. On the first access
+    * a connection will be tried to become established. After that
+    * the instance keeps the connection open until a certain timeout
+    * is reached. (Haven't found the exact number yet.) For further
```

Comment:
good comment.
I think the default connection timeout is 2 seconds according to the link
```
connect-timeout = 2 seconds
```

Reply:
thanks. :) 
I thought, that configuration is more about establishing the connection. As it's header is:  "# Maximum amount of time allowed to establish a connection to the Redis server". But I'm not sure. Will research a bit more later..